src/util-inl.h

```diff
@@ -198,6 +198,35 @@ TypeName* Unwrap(v8::Local<v8::Object> object) {
   return static_cast<TypeName*>(pointer);
 }
 
+template <typename TypeName>
+void SwapBytes(TypeName* dst,
+               const TypeName* src,
+               size_t buflen) {
+  int t_size = sizeof(TypeName);
+
+  // reinterpret src as unsigned because we want unsigned shifts
+  for (size_t i = 0; i < buflen; i++) {
+    switch (t_size) {
+    #define TOU16 static_cast<uint16_t>(src[i])
+    #define TOU32 static_cast<uint32_t>(src[i])
+    #define TOU64 static_cast<uint64_t>(src[i])
+
+    case 2: dst[i] = (TOU16 << 8) | (TOU16 >> 8); break;
+    case 4: dst[i] = (TOU32 << 24) | ((((TOU32 << 16) >> 16) >> 8) << 16)
+      | ((((TOU32 >> 16) << 16) << 8) >> 16) | (TOU32 >> 24);
+    break;
+    case 8: dst[i] = (TOU64 << 56) | ((((TOU64 << 48) >> 48) >> 8) << 48)
+      | ((((TOU64 >> 48) << 48) << 8) >> 48)
+      | ((((TOU64 << 40) >> 40) >> 16) << 40)
+      | ((((TOU64 >> 40) << 40) << 16) >> 40)
+      | ((((TOU64 << 32) >> 32) >> 24) << 32)
+      | ((((TOU64 >> 32) << 32) << 24) >> 32) | (TOU64 >> 56);
+    break;
+    }
+  }
+}
```

Comment:
This is much too obtuse to my liking.  I'd just restrict it to a non-template version that does only `uint16_t` and keeps the simple byte swapping logic.